tests/integration_tests/bulk_import/test_bulk_import.py

```diff
@@ -149,6 +150,53 @@ def test_import_data_rollback(
     assert actual_collection is None
 
 
+@pytest.mark.s3
+def test_bulk_import_saves_default_number_of_documents_if_no_limit_provided_in_request(
+    data_db: Session, client: TestClient, superuser_header_token
+):
+    input_json = build_json_file(
+        {
+            "collections": [default_collection],
+            "families": [default_family],
+            "documents": [
+                {**default_document, "import_id": f"test.new.document.{i}"}
+                for i in range(DEFAULT_DOCUMENT_LIMIT + 1)
+            ],
+            "events": [default_event],
+        }
+    )
+
+    response = client.post(
+        "/api/v1/bulk-import/UNFCCC.corpus.i00000001.n0000",
+        files={"data": input_json},
+        headers=superuser_header_token,
+    )
+
+    assert response.status_code == status.HTTP_202_ACCEPTED
+    assert response.json() == {
+        "message": "Bulk import request accepted. Check Cloudwatch logs for result."
+    }
+
+    assert (
+        "Created"
+        == data_db.query(FamilyDocument)
+        .filter(
+            FamilyDocument.import_id
+            == f"test.new.document.{DEFAULT_DOCUMENT_LIMIT - 1}"
+        )
+        .one_or_none()
+        .document_status
+    )
+
+    assert (
+        not data_db.query(FamilyDocument)
+        .filter(
+            FamilyDocument.import_id == f"test.new.document.{DEFAULT_DOCUMENT_LIMIT}"
```

Comment:
I'm not sure I fully understand your questions but, at the beginning of the test, we we generate a json file with the number of documents being DEFAULT_DOCUMENT_LIMIT+1. We then use the value of DEFAULT_DOCUMENT_LIMIT to establish that the last doc in the json which would be over the default limit was not saved to the DB. We do use the DEFAULT_DOCUMENT_LIMIT constant to generate the ids of the docs as well, with the idea being that we don't have to update the test if the value of the constant changes. But if you can see issues with this coupling, let me know. Maybe I missed something...

Reply:
Yup - that's exactly what I am asking. So we are using the ID to infer how many documents have been ingested.

My question was - could we be more explicit and use a `count` or something similar to answer "Did we only do X documents?".

For example - I might say (pseudo code):
```python
expected_ingested_doc_count = 1000
current_doc_count = db.getDocCount
ingest()
new_doc_count = db.getDocCount
ingested_doc_count = new_doc_count - current_doc_count
assert ingested_doc_count == expected_ingested_doc_count
```

Reply:
I can't think of a scenario where this would make the test flaky (but please let me know if you can). To be fair, the previous iteration of this test had the value hardcoded rather than using the constant. I thought that the latter would make the test clearer in specifying where the number comes from and also avoid us having to update the test if we change the constant in the future (i.e. decoupling the test from the implementation). Happy to revert it to the previous approach if you think it's better practice though.

Reply:
All good! the implementation works, and the tests captures the need.

The very slim use can I can find is if we change the ID generator to use something different e.g. hashes, rather than a counter.